addons/l10n_in_hr_payroll/models/l10n_in_hr_payroll.py

```diff
@@ -1,246 +1,193 @@
-                payslip_pool.write(cr, uid, slip_ids, {'advice_id': advice.id}, context=context)
-        return True
-
-    def confirm_sheet(self, cr, uid, ids, context=None):
+                payslip_line = self.env['hr.payslip.line'].search([('slip_id', '=', slip.id), ('code', '=', 'NET')], limit=1)
+                if payslip_line:
+                    self.env['hr.payroll.advice.line'].create({
+                        'advice_id': advice.id,
+                        'name': slip.employee_id.bank_account_id.acc_number,
+                        'employee_id': slip.employee_id.id,
+                        'bysal': payslip_line.total
+                    })
+                slip.advice_id = advice.id
+
+    @api.multi
+    def confirm_sheet(self):
         """
         confirm Advice - confirmed Advice after computing Advice Lines..
-        @param cr: the current row, from the database cursor,
-        @param uid: the current user’s ID for security checks,
-        @param ids: List of confirm Advice’s IDs
-        @return: confirmed Advice lines and set sequence of Advice.
-        @param context: A standard dictionary for contextual values
         """
-        seq_obj = self.pool.get('ir.sequence')
-        for advice in self.browse(cr, uid, ids, context=context):
+        for advice in self:
             if not advice.line_ids:
                 raise UserError(_('You can not confirm Payment advice without advice lines.'))
-            advice_date = datetime.strptime(advice.date, DATETIME_FORMAT)
-            advice_year = advice_date.strftime('%m') + '-' + advice_date.strftime('%Y')
-            number = seq_obj.next_by_code(cr, uid, 'payment.advice')
-            sequence_num = 'PAY' + '/' + advice_year + '/' + number
-            self.write(cr, uid, [advice.id], {'number': sequence_num, 'state': 'confirm'}, context=context)
-        return True
-
-    def set_to_draft(self, cr, uid, ids, context=None):
+            date = fields.Date.from_string(fields.Date.today())
+            advice_year = date.strftime('%m') + '-' + date.strftime('%Y')
+            number = self.env['ir.sequence'].next_by_code('payment.advice')
+            advice.number = 'PAY' + '/' + advice_year + '/' + number
+            advice.state = 'confirm'
+
+    @api.multi
+    def set_to_draft(self):
         """Resets Advice as draft.
         """
-        return self.write(cr, uid, ids, {'state':'draft'}, context=context)
+        self.ensure_one()
+        self.state = 'draft'
 
-    def cancel_sheet(self, cr, uid, ids, context=None):
+    @api.multi
+    def cancel_sheet(self):
         """Marks Advice as cancelled.
         """
-        return self.write(cr, uid, ids, {'state':'cancel'}, context=context)
+        self.ensure_one()
+        self.state = 'cancel'
 
-    def onchange_company_id(self, cr, uid, ids, company_id=False, context=None):
+    @api.multi
+    def onchange_company_id(self, company_id=False):
```

Comment:
This is not how we define an onchange in newAPI + The onchange is still in the view.